### PR TITLE
RDISCROWD-6222 Show GIGwork users why no tasks are available

### PR DESCRIPTION
**Pull request by peterkle**
*Issue number of the reported bug or feature request: #RDISCROWD-6222*

**Describe your changes**
As a user, I would like to be able to see why no more tasks are available to me even when a project is not yet completed, so that I can understand if there is a problem with the project or if I can work on more tasks.

A typical cause for this is that the project is using the User Preference or Task Queues scheduler and the remaining tasks have preferences configured that are different than the user's.

**Notes**
* Notice that I refactored the "redirect_to_password" check by moving it earlier in the function to avoid needless computation of metrics. This change is unrelated to this feature.
* JSON body shape:
```
  "notifications": {
    "project_incomplete_info": {
      "user_preferences": {
        "account_profile_link": "/account/user/", 
        "task_scheduler": "User Preference Scheduler"
      }
    }
  }
  ```

### CODE REVIEW

File: test/test_web.py [hunk 1]
```diff
@@ -1958,6 +1959,38 @@ def test_get_project_json(self):
...
+        admin.user_pref = {'language': ['en']}
+        project = ProjectFactory.create(owner=admin, info={'sched': 'default'})
+        task_user_pref = dict(languages=['sp'])
+        TaskFactory.create(project=project, user_pref=task_user_pref)
+        self.signin_user(admin)
+        url = '/project/%s/' % project.short_name
+
+        res = self.app_get_json(url)
+        data = json.loads(res.data)
+        assert 'notifications' not in data.keys(), list(data.keys())
```
Comment (kbecker42) [inline test/test_web.py:1975]:
```
# Default task scheduler: verify notification message does not display.
```

File: test/test_web.py [hunk 2]
```diff
@@ -1958,6 +1959,38 @@ def test_get_project_json(self):
...
+
+        project.info['sched'] = Schedulers.task_queue
+        res = self.app_get_json(url)
+        data = json.loads(res.data)
+        assert 'notifications' in data.keys(), list(data.keys())
```
Comment (kbecker42) [inline test/test_web.py:1980]:
```
# Task Queue scheduler: verify notification message displays.
```

Also, is this updating the project repo and saving the new task queue preference via `self.app_get_json(url)`?

File: test/test_web.py [hunk 3]
```diff
@@ -1958,6 +1959,38 @@ def test_get_project_json(self):
...
+
+        project.info['sched'] = Schedulers.user_pref
+        res = self.app_get_json(url)
+        data = json.loads(res.data)
+        assert 'notifications' in data.keys(), list(data.keys())
```
Comment (kbecker42) [inline test/test_web.py:1985]:
```
# User Preferences task scheduler: verify notification message displays.
```

**Review by kbecker42 — APPROVED** [review]
Thank you.

File: test/test_web.py (hunk 2 above)
Comment (peterkle) [reply test/test_web.py:1982]: `self.app_get_json(url)` is a test helper which only fetches data. The update to project scheduler is done via assignment in `project.info['sched'] = Schedulers.task_queue`

File: pybossa/view/projects.py [hunk 4]
```diff
@@ -1005,6 +1018,8 @@ def details(short_name):
         template_args['ckan_name'] = current_app.config.get('CKAN_NAME')
         template_args['ckan_url'] = current_app.config.get('CKAN_URL')
         template_args['ckan_pkg_name'] = short_name
+    if len(notifications):
```
Comment (XiChenn) [inline pybossa/view/projects.py:1021]: Minor. If you prefer a more Pythonic way, you can use ```if not notifications:```.

**Review by XiChenn — APPROVED** [review]
lgtm

**Review by XiChenn — APPROVED** [review]
Feel free to "Squash and merge" upon all checks passed.

**Pull request merged**